server/webapp/WEB-INF/rails.new/webpack/views/agents/agent_row_widget.js.msx

```diff
@@ -61,9 +61,14 @@ const AgentRowWidget = {
     const environments    = joinOrNoneSpecified(agent.environments());
     let agentStatus       = agent.status();
     const isBuildingAgent = !agent.buildDetails().isEmpty();
+    const isElasticAgent  = agent.isElasticAgent();
     let selectAgentCheckbox;
     let hostnameLink      = <span>{agent.hostname()}</span>;
 
+    if (isElasticAgent) {
+      agent.fetchElasticAgentIcon();
```

Comment:
This has another drawback that a redraw (because of a user action) before the ajax call finishes will cause a second ajax call possibly at a 60hz frequency and totally overwhelm the server.